merlin/models/tf/core/transformations.py

```diff
@@ -662,45 +663,67 @@ class HashedCross(TabularBlock):
+            "one_hot", configuring the layer as follows:
             - `"int"`: Return the integer bin indices directly.
-            - `"one_hot"`: Encodes each individual element in the input into an
-                array the same size as `num_bins`, containing a 1 at the input's bin
-                index.
-        sparse : bool
-            Boolean. Only applicable to `"one_hot"` mode. If True, returns a
-            `SparseTensor` instead of a dense `Tensor`. Defaults to False.
-        output_name : string
+            - `"one_hot"`: Encodes each individual element in the input into an array with the same
+                size as `num_bins`, containing a 1 at the input's bin index.
+        sparse: bool
+            Boolean. Only applicable to `"one_hot"` mode. If True, returns a `SparseTensor` instead
+            of a dense `Tensor`. Defaults to False.
+        output_name: string
             Name of output feature, if not specified, default would be
             cross_<feature_name>_<feature_name>_<...>
+        inffer_num_bins: bool
+            If True, num_bins would be set as the multiplier of feature cadinalities, if the
+            multiplier is bigger than max_num_bins, then it would be cliped by max_num_bins
+        max_num_bins: int
+            Upper bound of num_bins, by default 100000.
     """
 
     def __init__(
         self,
         schema: Schema,
-        num_bins: int,
+        num_bins: int = None,
         sparse: bool = False,
         output_mode: str = "int",
         output_name: str = None,
+        inffer_num_bins: bool = False,
+        max_num_bins: int = 100000,
         **kwargs,
     ):
         super().__init__(**kwargs)
 
+        if (not inffer_num_bins) and (num_bins is None):
+            raise ValueError(
+                "num_bins is not given, and inffer_num_bins is False, either of them "
+                "is required, if you want to set fixed num_bins, then set inffer_num_bins to False,"
+                " and set num_bins to an integer value, if you want to inffer num_bins from the "
+                "mulplier of feature cardinalities, at the same time you can set the max_num_bins."
+            )
+
         if not (output_mode in ["int", "one_hot"]):
             raise ValueError("output_mode must be 'int' or 'one_hot'")
         self.schema = schema
-        self.num_bins = num_bins
         self.output_mode = output_mode
         self.sparse = sparse
         if not output_name:
             self.output_name = "cross"
-            for name in self.schema.column_names:
+            for name in schema.column_names:
                 self.output_name = self.output_name + "_" + name
         else:
             self.output_name = output_name
 
+        # Set num_bins
+        if num_bins:
+            self.num_bins = num_bins
+        else:
+            cardinalities = schema_utils.categorical_cardinalities(schema)
+            multiplier = 1
```

Comment:
Hi @gabrielspmoreira, I think I have to keep it as an integer, since all the cardinalities are integers, and it will be used as the shape, tensorflow requires that the values of shape should be integer, or it would raise `TypeError: Dimension value must be integer or None or have an __index__ method, got value '9.0' with type '<class 'float'>'`

Reply:
Fixed, thank you.